.github/workflows/PR.yml

```diff
@@ -9,7 +9,20 @@ on:
       - 'mkdocs.yml'
 
 jobs:
+  spotless:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-java@v3.12.0
+        with:
+          distribution: 'zulu'
+          java-version-file: .github/workflows/.ci-java-version
+      - run: ./gradlew spotlessCheck
+
   build:
+    needs: spotless
```

Comment:
I don't think we need a strict dependency here. If the spotless check job fails then it will auto-cancel all of the other jobs (once we add spotless as a required check)